[bug] 1.5.0 kcm-regional-telemetry crashes

* Part of https://github.com/k0rdent/kof/issues/603
* Deployed KCM Region by docs.
* One of KCM pods is crashing in the mgmt cluster:
```
kubectl get pod -A | grep CrashLoopBackOff

kcm-system           kcm-regional-telemetry-8465bf9f99-5m5r8                           0/1     CrashLoopBackOff   7 (3m30s ago)   82m
```
* Logs:
```
kubectl logs -n kcm-system deploy/kcm-regional-telemetry

{"level":"info","ts":"2025-11-07T14:15:21Z","logger":"setup","msg":"starting manager"}
{"level":"info","ts":"2025-11-07T14:15:21Z","logger":"controller-runtime.metrics","msg":"Starting metrics server"}
{"level":"info","ts":"2025-11-07T14:15:21Z","logger":"controller-runtime.metrics","msg":"Serving metrics server","bindAddress":"127.0.0.1:8080","secure":false}
{"level":"info","ts":"2025-11-07T14:15:21Z","msg":"starting server","name":"health probe","addr":"[::]:8081"}
I1107 14:15:21.264011       1 leaderelection.go:257] attempting to acquire leader lease kcm-system/7f46165474d11ee5836777d85df2cdab.k0rdent.mirantis.com...
I1107 14:15:38.395749       1 leaderelection.go:271] successfully acquired lease kcm-system/7f46165474d11ee5836777d85df2cdab.k0rdent.mirantis.com
{"level":"info","ts":"2025-11-07T14:15:38Z","logger":"telemetry-runner","msg":"Starting telemetry runner","interval":3600,"jitter":0.1}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0xec5210]
goroutine 168 [running]:
github.com/K0rdent/kcm/internal/telemetry/collector.(*SegmentIO).Collect.func1()
	github.com/K0rdent/kcm/internal/telemetry/collector/segmentio.go:107 +0x120
created by github.com/K0rdent/kcm/internal/telemetry/collector.(*SegmentIO).Collect in goroutine 61
	github.com/K0rdent/kcm/internal/telemetry/collector/segmentio.go:101 +0x350
```

--- a/internal/telemetry/collector/data_fetcher.go
+++ b/internal/telemetry/collector/data_fetcher.go
@@ -102,12 +102,13 @@ func (f *parentDataFetcher) fetch(ctx context.Context, parentClient client.Clien
 			return fmt.Errorf("failed to list ClusterDeployments: %w", err)
 		}
 
-		f.clusters = make([]client.Object, len(clds.Items))
-		for i, v := range clds.Items {
+		f.clusters = make([]client.Object, 0, len(clds.Items))
+		for _, v := range clds.Items {
 			if v.Status.Region != "" {
 				continue
 			}
-			f.clusters[i] = &v
+
+			f.clusters = append(f.clusters, &v)
 		}
 	}
 

--- a/internal/telemetry/collector/segmentio.go
+++ b/internal/telemetry/collector/segmentio.go
@@ -95,6 +95,10 @@ func (t *SegmentIO) Collect(ctx context.Context) error {
 	}
 
 	for _, cluster := range parentData.clusters {
+		if cluster == nil { // sanity check
+			continue
+		}
+
 		wg.Add(1)
 		sem <- struct{}{}
 

